### PR TITLE
Add GCFv2 Storage sample

### PR DESCRIPTION
**Pull request by averikitsch**
Fixes #issue

> It's a good idea to open an issue first for discussion.

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Comment by eaball35** [comment]
```
#
# functions/v2/helloworld/hello-gcs
#

[ERROR] src/main/java/functions/HelloGcs.java:[39] (sizes) LineLength: Line is longer than 100 characters (found 119).
[ERROR] src/main/java/functions/HelloGcs.java:[41,7] (indentation) Indentation: 'event' has incorrect indentation level 6, expected level should be 8.
[ERROR] src/main/java/functions/HelloGcs.java:[42,7] (indentation) Indentation: 'PojoCloudEventDataMapper' has incorrect indentation level 6, expected level should be 8.
[ERROR] src/main/java/functions/HelloGcs.java:[42,49] (whitespace) WhitespaceAfter: ',' is not followed by whitespace.
[ERROR] src/test/java/functions/HelloGcsTest.java:[59,18] (naming) LocalVariableName: Local variable name 'Obj' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'.
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (default-cli) on project functions-hello-gcs: You have 5 Checkstyle violations. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException


err: exit status 1
```

File: functions/v2/helloworld/hello-gcs/src/main/java/functions/HelloGcs.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  @Override
+  public void accept(CloudEvent event) {
+    logger.info("Event: " + event.getId());
+    logger.info("Event Type: " + event.getType());
+
+    if (event.getData() != null) {
+      String cloudEventData = new String(event.getData().toBytes(), StandardCharsets.UTF_8);
+      Gson gson = new Gson();
+      GcsEvent gcsEvent = gson.fromJson(cloudEventData, GcsEvent.class);
+      logger.info("Bucket: " + gcsEvent.getBucket());
```
Comment (ace-n) [inline functions/v2/helloworld/hello-gcs/src/main/java/functions/HelloGcs.java:39]: Nit: add a newline before this one? (so the `logger.info` statements are separate from event marshaling code).

File: functions/v2/helloworld/hello-gcs/src/main/java/functions/eventpojos/GcsEvent.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package functions.eventpojos;
+
+import java.util.Date;
+
+public class GcsEvent {
```
Comment (ace-n) [inline functions/v2/helloworld/hello-gcs/src/main/java/functions/eventpojos/GcsEvent.java:21]: I assume that the CloudEvent type libraries won't work here?

(IIUC, they were supposed to obviate the need for these POJOs.)

File: functions/v2/helloworld/hello-gcs/src/test/java/functions/HelloGcsTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  @After
+  public void afterTest() {
+    LOG_HANDLER.clear();
+  }
+
+  @Test
+  public void helloGcs_shouldPrintFileName() throws JsonProcessingException {
+    // Create event data
+    GcsEvent gcsEvent = new GcsEvent();
```
Comment (ace-n) [inline functions/v2/helloworld/hello-gcs/src/test/java/functions/HelloGcsTest.java:52]: Nit: I wonder if the CloudEvent type libraries (if we used them) would support the Builder pattern here.

File: functions/v2/helloworld/hello-gcs/src/test/java/functions/HelloGcsTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            .withId("0")
+            .withType("google.storage.object.finalize")
+            .withSource(URI.create("https://example.com"))
+            .withData("application/json", jsonData.getBytes())
+            .build();
+
+    new HelloGcs().accept(event);
+
+    String message = LOG_HANDLER.getStoredLogRecords().get(3).getMessage();
+    assertThat(message).contains("File: foo.txt");
```
Comment (ace-n) [inline functions/v2/helloworld/hello-gcs/src/test/java/functions/HelloGcsTest.java:75]: Nit: do we want to assert on the other properties as well?

**Review by ace-n — APPROVED** [review]
LGTM other than a few nits.

File: functions/v2/helloworld/hello-gcs/src/main/java/functions/eventpojos/GcsEvent.java (hunk 2 above)
Comment (averikitsch) [reply functions/v2/helloworld/hello-gcs/src/main/java/functions/eventpojos/GcsEvent.java:21]: We aren't using the libraries. They are not GA

File: functions/v2/helloworld/hello-gcs/src/test/java/functions/HelloGcsTest.java (hunk 3 above)
Comment (averikitsch) [reply functions/v2/helloworld/hello-gcs/src/test/java/functions/HelloGcsTest.java:52]: I don't believe the library is has that functionality either

**Pull request merged**